File: ml/src/main/scala/ai/h2o/sparkling/ml/params/HasBetaConstraints.scala
package ai.h2o.sparkling.ml.params

import ai.h2o.sparkling.H2OFrame
import org.apache.spark.sql.DataFrame
import ai.h2o.sparkling.utils.DataFrameSerializationWrappers._

trait HasBetaConstraints extends H2OAlgoParamsBase with HasDataFrameSerializer {
  private val betaConstraints = new NullableDataFrameParam(
    this,
    "betaConstraints",
    "Data frame of beta constraints enabling to set special conditions over the model coefficients.")

  setDefault(betaConstraints -> null)

  def getBetaConstraints(): DataFrame = $(betaConstraints)

  def setBetaConstraints(value: DataFrame): this.type = set(betaConstraints, toWrapper(value))

  private[sparkling] def getBetaConstraintsParam(trainingFrame: H2OFrame): Map[String, Any] = {
    Map("beta_constraints" -> convertDataFrameToH2OFrameKey(getBetaConstraints()))
  }

  override private[sparkling] def getSWtoH2OParamNameMap(): Map[String, String] = {
    super.getSWtoH2OParamNameMap() ++ Map("betaConstraints" -> "beta_constraints")
  }
}

File: py-scoring/src/ai/h2o/sparkling/ml/params/HasBetaConstraints.py
from ai.h2o.sparkling.ml.params.H2OTypeConverters import H2OTypeConverters
from pyspark.ml.param import *


class HasBetaConstraints(Params):
    betaConstraints = Param(
        Params._dummy(),
        "betaConstraints",
        "Data frame of beta constraints enabling to set special conditions over the model coefficients.",
        H2OTypeConverters.toNullableDataFrame())

    def getBetaConstraints(self):
        return self.getOrDefault(self.betaConstraints)

    def setBetaConstraints(self, value):
        return self._set(betaConstraints=value)
